modules/core/shared/src/main/scala/gem/Step.scala

```diff
@@ -16,32 +19,140 @@ object Step {
-  sealed case class Nici(dynamicConfig: DynamicConfig.Nici, base: Base) extends Step
-  sealed case class Nifs(dynamicConfig: DynamicConfig.Nifs, base: Base) extends Step
-  sealed case class Gpi(dynamicConfig: DynamicConfig.Gpi, base: Base) extends Step
-  sealed case class Gsaoi(dynamicConfig: DynamicConfig.Gsaoi, base: Base) extends Step
-  sealed case class GmosS(dynamicConfig: DynamicConfig.GmosS, base: Base) extends Step
   sealed case class AcqCam(dynamicConfig: DynamicConfig.AcqCam, base: Base) extends Step
-  sealed case class GmosN(dynamicConfig: DynamicConfig.GmosN, base: Base) extends Step
   sealed case class Bhros(dynamicConfig: DynamicConfig.Bhros, base: Base) extends Step
-  sealed case class Visitor(dynamicConfig: DynamicConfig.Visitor, base: Base) extends Step
   sealed case class Flamingos2(dynamicConfig: DynamicConfig.Flamingos2, base: Base) extends Step
   sealed case class Ghost(dynamicConfig: DynamicConfig.Ghost, base: Base) extends Step
+  sealed case class Gpi(dynamicConfig: DynamicConfig.Gpi, base: Base) extends Step
+  sealed case class GmosN(dynamicConfig: DynamicConfig.GmosN, base: Base) extends Step
+  sealed case class GmosS(dynamicConfig: DynamicConfig.GmosS, base: Base) extends Step
+  sealed case class Gnirs(dynamicConfig: DynamicConfig.Gnirs, base: Base) extends Step
+  sealed case class Gsaoi(dynamicConfig: DynamicConfig.Gsaoi, base: Base) extends Step
+  sealed case class Michelle(dynamicConfig: DynamicConfig.Michelle, base: Base) extends Step
+  sealed case class Nici(dynamicConfig: DynamicConfig.Nici, base: Base) extends Step
+  sealed case class Nifs(dynamicConfig: DynamicConfig.Nifs, base: Base) extends Step
+  sealed case class Niri(dynamicConfig: DynamicConfig.Niri, base: Base) extends Step
+  sealed case class Phoenix(dynamicConfig: DynamicConfig.Phoenix, base: Base) extends Step
+  sealed case class Trecs(dynamicConfig: DynamicConfig.Trecs, base: Base) extends Step
+  sealed case class Visitor(dynamicConfig: DynamicConfig.Visitor, base: Base) extends Step
+
+
+  object GmosN extends GmosNOptics
+
+  trait GmosNOptics {
+
+    /** @group Optics */
+    val dynamicConfig: Lens[Step.GmosN, DynamicConfig.GmosN] =
+      Lens[GmosN, DynamicConfig.GmosN](_.dynamicConfig)(a => _.copy(dynamicConfig = a))
+
+    /** @group Optics */
+    val base: Lens[GmosN, Base] =
+      Lens[GmosN, Base](_.base)(a => _.copy(base = a))
+
+  }
+
+  object GmosS extends GmosSOptics
+
+  trait GmosSOptics {
+
+    /** @group Optics */
+    val dynamicConfig: Lens[Step.GmosS, DynamicConfig.GmosS] =
+      Lens[GmosS, DynamicConfig.GmosS](_.dynamicConfig)(a => _.copy(dynamicConfig = a))
+
+    /** @group Optics */
+    val base: Lens[GmosS, Base] =
+      Lens[GmosS, Base](_.base)(a => _.copy(base = a))
+
+  }
 
-  implicit val EqStep: Eq[Step] =
-    Eq.fromUniversalEquals
+  implicit def EqStep[S <: Step]: Eq[S] =
+    Eq.by(s => (s.dynamicConfig, s.base))
```

Comment:
This may be sketchy, idk.

Reply:
It seems fine to me

Reply:
In three years, when a new kind of `Step` is needed with a different type of equality, will your future self hate you ?
IMO it is a tiny bit sketchy, but the base trait is sealed, so at least when someone needs to add a new type of Step in the future, he/she will find this quickly.